src/main/java/uk/gov/hmcts/darts/task/service/impl/ManualDeletionProcessorImpl.java

```diff
@@ -100,15 +149,26 @@ private void deleteFromExternalDataStore(ExternalObjectDirectoryEntity externalO
         }
     }
 
-    private boolean isTranscriptionNotDeleted(ObjectAdminActionEntity action) {
+    boolean isTranscriptionNotDeleted(ObjectAdminActionEntity action) {
         return action.getTranscriptionDocument() != null && !action.getTranscriptionDocument().isDeleted();
     }
 
-    private boolean isMediaNotDeleted(ObjectAdminActionEntity action) {
+    boolean isMediaNotDeleted(ObjectAdminActionEntity action) {
         return action.getMedia() != null && !action.getMedia().isDeleted();
     }
 
-    public OffsetDateTime getDeletionThreshold() {
-        return OffsetDateTime.now().minus(Duration.parse("PT" + gracePeriod));
+    OffsetDateTime getDeletionThreshold() {
+        return OffsetDateTime.now().minus(gracePeriod);
+    }
+
+    static class RetConfReason {
```

Comment:
Any reason why the JSON structure is defined as inner class? Cant we have it as a separate class similar to CaseRetentionConfidenceReason? The both (old and new) JSON gets passed into the Updatemetadata;s confidence reason field. One is associated with Case retention and the new one is for Manual Deletion.

Reply:
Yeah this class is exclusively used within the ManualDeletionProcessor as we convert it into a string before we send it to other classes. As such it makes sense to keep the logic centralised as the RetConfReason inner class only has any meaning when combined with the ManualDeletionProcessor.

If this changes in the future I think we can pull it out but I do not see any value in doing so now as the class is meaningless without the ManualDeletion context. (So improves maintainability)

Reply:
We already have a pattern of having the json objects separate from the calling classes. You are breaking this pattern by using an inner class

Reply:
The purpose of nested classes is to group classes that belong together, which makes your code more readable and maintainable and is standard practice in Java development.

As this class is exclusively used within the ManualDeletionProcessor and has no use / context outside of this class there no value in moving it into its own class as we lose the benefits outlined above.

If the existing pattern we are using goes against these standards the solution should be to refactor those classes instead of enforcing them to ensure we produce quality code

Reply:
Does it not increase the line of Code of a main class by adding a inner class for an object which defines a Json structure?

Reply:
The class it is added in is a relatively small in nature anyway so the additional  9 lines of code wont have any impact but it will make maintainability easier as the inner class has no meaning outside of the ManualDeletion class so having this logic in the same class helps provide context

Reply:
I think this is deviating from what we have done already and introducing a new pattern. If we are changing the code i.e. maintaining the JSON template as inner class rather than a separate class, then we will end up introducing 2 different patter in the code. I dont see any issue with the existing pattern.

Reply:
How come maintaining a JSON template in a separate class, which makes it reusable in future and reduces the line of code in main class, is a TechDebt? I do agree, from encapsulation point of view, you want to maintain everything in the same class but its more of a trade off. Considering Easy maintenance, More readable (if the Json template grows), Reusable etc we should have a separate class.